scripts/add_existing_baseyear.py

```diff
@@ -0,0 +1,394 @@
+    ratio_residential=pd.Series([(n.loads_t.p_set.sum()['{} residential rural heat'.format(node)] /
+               (n.loads_t.p_set.sum()['{} residential rural heat'.format(node)] +
+                n.loads_t.p_set.sum()['{} services rural heat'.format(node)] ))
+                   for node in df.index], index=df.index)
+       
+    for tech in techs: 
+        df['residential ' + tech] = df[tech]*ratio_residential
+        df['services ' + tech] = df[tech]*(1-ratio_residential)
+               
+    nodes={}
+    p_nom={}        
+    for name in ["residential rural", 
+                 "services rural",
+                 "residential urban decentral",
+                 "services urban decentral", 
+                 "urban central"]:
+            
+        name_type = "central" if name == "urban central" else "decentral"            
+        nodes[name] = pd.Index([index[0:5] for index in n.buses.index[n.buses.index.str.contains(name) & n.buses.index.str.contains('heat')]])
+        heat_pump_type = "air" if "urban" in name else "ground"
+        heat_type= "residential" if "residential" in name else "services"
+            
+        if name == "urban central":
+            p_nom[name]=df['air heat pump'][nodes[name]]
+        else:
+            p_nom[name] =  df['{} {} heat pump'.format(heat_type, heat_pump_type)][nodes[name]]
+            
+        # Add heat pumps        
+        costs_name = "{} {}-sourced heat pump".format("decentral", heat_pump_type)
+
+
+        cop = {"air" : ashp_cop, "ground" : gshp_cop}
+        efficiency = cop[heat_pump_type][nodes[name]] if time_dep_hp_cop else costs.at[costs_name,'efficiency']
+
+        n.madd("Link",
+               nodes[name],
+               suffix=" {} {} heat pump <{}".format(name,heat_pump_type, baseyear),
+               bus0=nodes[name],
+               bus1=nodes[name] + " " + name + " heat",
+               carrier="{} {} heat pump".format(name,heat_pump_type),
+               efficiency=efficiency,
+               capital_cost=costs.at[costs_name,'efficiency']*costs.at[costs_name,'fixed'],
+               p_nom=p_nom[name]*max(0,(2045-int(year))/25)) #decomissioning happens lineary from now to 25 years lter  
+           
+        # add resistive heater, gas boilers and oil boilers 
+        # (50% capacities to rural buses, 50% to urban buses)
+        n.madd("Link",
+               nodes[name],
+               suffix= " " + name + " resistive heater <{}".format(baseyear),
+               bus0=nodes[name],
+               bus1=nodes[name] + " " + name + " heat",
+               carrier=name + " resistive heater",
+               efficiency=costs.at[name_type + ' resistive heater','efficiency'],
+               capital_cost=costs.at[name_type + ' resistive heater','efficiency']*costs.at[name_type + ' resistive heater','fixed'],
+               p_nom=0.5*df['{} resistive heater'.format(heat_type)][nodes[name]]*max(0,(2045-int(year))/25))
```

Comment:
The years 2045 and lifetime assumptions should go in the config.yaml.

Reply:
Added in https://github.com/PyPSA/pypsa-eur-sec/pull/63/commits/2b3dc60f11179d4efd3cd3a4f11f86fe679dab9c and https://github.com/PyPSA/pypsa-eur-sec/pull/63/commits/a4c22a29a43f63e52d879267a7f26a5d27b5faa5